Improve roxygen highlighting

jranke opened the pull request:
Following the discussion in jalvesaq/Nvim-R#84
- Add support for titles spanning more than one row
- Add support for roxygen blocks without a title, as this can make sense
  when an @rdname tag is present
- If no title is recognised, and no @rdname is present, just highlight
  as R comment section

jranke committed:
Improve roxygen highlighting

Following the discussion in jalvesaq/Nvim-R#84
- Add support for titles spanning more than one row
- Add support for roxygen blocks without a title, as this can make sense
  when an @rdname tag is present
- If no title is recognised, and no @rdname is present, just highlight
  as R comment section

klmr commented:
Thanks for doing the work, I’ll try it out later today and get back to you.

klmr committed:
Add failing Roxygen highlight tests

klmr reviewed (commented):
Great effort, and it works very well for most situations. However, please see https://github.com/jranke/R-Vim-runtime/pull/1 for more test cases where the changes fail.

klmr committed:
Remove examples that contradict Roxygen doc

Roxygen documentation and implementation conflict; the implementation in
roxygen2 accepts the deleted test cases but according to the
documentation they are invalid. So they are removed.

jranke committed:
Merge pull request #1 from klmr/roxygen-highlight-tests

Add failing Roxygen highlight test

jranke committed:
Further improve highlighting of roxygen tags

- Support roxygen blocks containing only a title. These are all roxygen
  blocks that do not contain an empty roxygen line nor a roxygen tag
- Add more test cases
- Only pure title blocks or titles followed by an empty roxygen line are
  recognised, although roxygen may recognize titles in some other cases,
  too

jranke commented:
@klmr Could you have another look? I have added support for pure title blocks and added a few more examples.

klmr committed:
Test case for Roxygen paragraph break

klmr commented:
You’re going to hate me, I found another failing case :-( https://github.com/jranke/R-Vim-runtime/pull/2

But I think this is the only remaining thing. In fact, the revised highlighting helped me find a genuine bug in some of my code (where a Roxygen block was interrupted and thus created a wrong documentation). The above PR is the only failing case I could find in any of my projects.

jranke committed:
Merge pull request #2 from klmr/roxygen-highlight-tests

Test case for Roxygen paragraph break

jranke committed:
Roxygen blocks with @rdname or @describeIn have no title

Also unify formatting of the roxygen highlighting tests

jranke commented:
OK, are we ready now?

klmr committed:
Add another paragraph break to make test general

klmr committed:
Make leading space in Roxygen line optional

Roxygen paragraph breaks are specified by an empty line except for the
`#'` leader; no space is needed.

klmr commented:
And another small adjustment. This time I’ve tried my hand at modifying the Vim code myself. Please have a look whether this is usable. — https://github.com/jranke/R-Vim-runtime/pull/3

jranke committed:
Merge pull request #3 from klmr/roxygen-highlight-tests

Remove required leading space

jranke commented:
@klmr Great! I think we are ready to go then.

@jalvesaq After some iterations we fixed a lot of cases were the roxygen highlighting failed to do the right thing. This pull request also adds a nice set of test cases.

jalvesaq commented:
Thanks!
